### DISCUSSION TITLE
Variables related to each other

### DISCUSSION THREAD

**Discussion by Jonix71** [rtbrick/bngblaster#166] [Q&A]
To avoid creating one access group for each emulated L2 device i need two counters related to each other.

counter1: 1-48
counter2: 1-1000

port-{counter1}-device{counter2}

When counter 1 reach max value in config counter2 should increase with 1.

so option 82 values would become

port-1-device1
..
port-48-device1

port1-device2
..
port 48-device2

port1-device3
..
port48-device3

and so on. Is this possible to do in BNGBlaster?

currently i need to create one access group for each L2 device it would make a large config file. 

           "access": [
             {
                "interface": "enp0s31f6",
                "type": "ipoe",
                "agent-remote-id": port{session}.device1",
                "agent-circuit-id": "port{session}.device1",
                "vlan-mode": "N:1",
                "stream-group": 1,
                "outer-vlan": 65,
                "monkey": true,
                "stream-group-id": 1
             },

Any sugestions?

**Comment by GIC-de** [comment]
**https://rtbrick.github.io/bngblaster/interfaces.html#triple-tagged**
_The configuration attributes for username, password, agent-remote-id, agent-circuit-id, and cfm-ma-name support variable substitution. The variable {session-global} will be replaced with the actual session-id starting from 1 and incremented for every new session. The variable {session} is incremented per-interface section. The variables {outer-vlan} and {inner-vlan} will be replaced with the corresponding VLAN identifier or 0 if not defined. The two variables {i1} and {i2} are configurable per-interface sections with user-defined start values and steps._

The existing iterators i1 and i2 may help but actually we need a further configuration like i1-count and i2-count to define how long to remain on a given value (how many sessions) before incrementing using iX-step. Would this help?
